src/Parallel/centrality/pagerank.jl

```diff
@@ -8,30 +8,35 @@ function pagerank(
     # indegree(g, v) is estimated run-time to iterate over inneighbors(g, v)
     partitions = LightGraphs.optimal_contiguous_partition(indegree(g), nthreads(), nv(g))
 
-    # collect dangling nodes
-    dangling_nodes = [v for v in vertices(g) if outdegree(g, v) == 0]
+    α_div_outdegree = Vector{Float64}(undef,nv(g))
+    dangling_nodes = Vector{U}()
+    for v in vertices(g)
+        if outdegree(g, v) == 0
+            push!(dangling_nodes, v)
+        end
+        α_div_outdegree[v] = (α/outdegree(g, v))
+    end
+
     N = Int(nv(g))
     # solution vector and temporary vector
     x = fill(1.0 / N, N)
     xlast = copy(x)
-    # personalization vector
-    p = fill(1.0 / N, N)
-    # adjustment for leaf nodes in digraph
-    dangling_weights = p
     for _ in 1:n
         dangling_sum = 0.0
         for v in dangling_nodes
             dangling_sum += x[v]
         end
         # flow from teleprotation
         for v in vertices(g)
-            xlast[v] = (1 - α + α * dangling_sum) * p[v]
+            xlast[v] = (1 - α + α * dangling_sum) * (1.0 / N)
         end
         # flow from edges
-        @threads for v_set in partitions
-            for v in v_set
-                for u in inneighbors(g, v)
-                    xlast[v] += α * x[u] / outdegree(g, u)
+        let x = x
+            @threads for v_set in partitions
+                for v in v_set
+                    for u in inneighbors(g, v)
```

Comment:
Is there a simple way, so that you could use `outneighbors` instead? That would be better for performance for types such as `SimpleWeightedDiGraph`

Reply:
I  don't see how Weighted graphs might be used here ? Don't we require a weight matrix for that ?

Reply:
No, what actually happens is that the default argument for `distmx` is `weights(g)`, and in the case of `SimpleWeightedDiGraph` this will actually be the matrix that is stored in this graph. Unfortunately this is also the reason why `inneighbors` is slower for this type of graph. But if you can't find a trivial way to replace it with `outneighbors` we maybe better leave it as it is.

Reply:
I could possibly do that but I have no knowledge if  https://github.com/JuliaGraphs/LightGraphs.jl/blob/275a20a79bf6a3f8241324caf4794f82f205fe08/src/Parallel/centrality/pagerank.jl#L9
this should affect it any way ?